unittest/abi_tests.cpp

```diff
@@ -0,0 +1,735 @@
+        BOOST_TEST("EVT6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV" == (std::string)upgrp2.group.get_leaf_key(son2_son0));
+        BOOST_TEST(1 == son2_son0.weight);
+
+        son2_son1 = upgrp2.group.get_child_node(son2, 1);
+        BOOST_TEST_REQUIRE(son2_son1.validate());
+        BOOST_TEST_REQUIRE(son2_son1.is_leaf());
+        BOOST_TEST("EVT8MGU4aKiVzqMtWi9zLpu8KuTHZWjQQrX475ycSxEkLd6aBpraX" == (std::string)upgrp2.group.get_leaf_key(son2_son1));
+        BOOST_TEST(2 == son2_son1.weight);
+
+        verify_type_round_trip_conversion<updategroup>(abis, "updategroup", var);
+    }
+    FC_LOG_AND_RETHROW()
+}
+
+BOOST_AUTO_TEST_CASE(newaccount_test) {
+    try {
+        abi_serializer abis(evt_contract_abi());
+
+        BOOST_CHECK(true);
+        const char* test_data = R"=====(
+        {
+          "name": "account",
+          "owner": ["EVT6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV"]
+        }
+        )=====";
+
+        auto var     = fc::json::from_string(test_data);
+        auto newacct = var.as<newaccount>();
+
+        BOOST_TEST("account" == newacct.name);
+        BOOST_TEST_REQUIRE(1 == newacct.owner.size());
+        BOOST_TEST("EVT6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV" == (std::string)newacct.owner[0]);
+
+        auto var2     = verify_byte_round_trip_conversion(abis, "newaccount", var);
+        auto newacct2 = var2.as<newaccount>();
+
+        BOOST_TEST("account" == newacct2.name);
+        BOOST_TEST_REQUIRE(1 == newacct2.owner.size());
+        BOOST_TEST("EVT6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV" == (std::string)newacct2.owner[0]);
+    }
+    FC_LOG_AND_RETHROW()
+}
+
+BOOST_AUTO_TEST_CASE(updateowner_test) {
+    try {
+        abi_serializer abis(evt_contract_abi());
+
+        BOOST_CHECK(true);
+        const char* test_data = R"=====(
+        {
+          "name": "account",
+          "owner": ["EVT6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV"]
+        }
+        )=====";
+
+        auto var    = fc::json::from_string(test_data);
+        auto upower = var.as<updateowner>();
+
+        BOOST_TEST("account" == upower.name);
+        BOOST_TEST_REQUIRE(1 == upower.owner.size());
+        BOOST_TEST("EVT6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV" == (std::string)upower.owner[0]);
+
+        auto var2    = verify_byte_round_trip_conversion(abis, "updateowner", var);
+        auto upower2 = var2.as<updateowner>();
+
+        BOOST_TEST("account" == upower2.name);
+        BOOST_TEST_REQUIRE(1 == upower2.owner.size());
+        BOOST_TEST("EVT6MRyAjQq8ud7hVNYcfnVPJqcVpscN5So8BhtHuGYqET5GDW5CV" == (std::string)upower2.owner[0]);
+    }
+    FC_LOG_AND_RETHROW()
+}
+
+BOOST_AUTO_TEST_CASE(transferevt_test) {
+    try {
+        abi_serializer abis(evt_contract_abi());
+
+        BOOST_CHECK(true);
+        const char* test_data = R"=====(
+        {
+          "from": "account1",
+          "to": "account2",
+          "amount": "12.0000 EVT"
```

Comment:
Update the code to the latest, the precision of the `EVT` symbol has changed from 4 to 5 :)